src/main/java/iguanaman/iguanatweakstconstruct/tweaks/RepairCraftingRecipe.java

```diff
@@ -23,18 +26,33 @@ public class RepairCraftingRecipe implements IRecipe {
                 "");
     }
 
-    private ModToolRepair modifier = null;
+    private ItemModifier modifier = null;
     private ItemStack modifiedTool = null;
 
     public RepairCraftingRecipe() {
-        for (ItemModifier mod : ModifyBuilder.instance.itemModifiers) if (mod instanceof ModToolRepair) {
-            modifier = (ModToolRepair) mod;
-            break;
+        boolean gregworksLoaded = Loader.isModLoaded("TGregworks");
+        for (ItemModifier mod : ModifyBuilder.instance.itemModifiers) {
+            // If TGregworks is present, use ModTGregRepair instead of ModToolRepair
+            if (gregworksLoaded && isTGregRepairModifier(mod)) {
```

Comment:
I forget the specific details of how/when class loading works -- seems fine if this works w/o tgregworks loaded, but please validate.  Seems fine since the instanceof check is in another method...

Reply:
Tested on a barebone forge install with only these mods loaded:
<img width="354" height="153" alt="image" src="https://github.com/user-attachments/assets/08a2ff91-8ad6-41cb-a66d-bf183e761db9" />

Repair still works as expected.
Also tested on 2.8.0-beta4, repair works as expected too